homeassistant/components/ambient_station/__init__.py

```diff
@@ -286,6 +286,35 @@
     return True
 
 
+async def async_migrate_entry(hass, config_entry):
+    """Migrate old entry."""
+    version = config_entry.version
+
+    _LOGGER.debug('Migrating from version %s', version)
+
+    reasons_to_re_add = {
+        1: "The format for unique IDs (in the entity registry) has changed."
+    }
+
+    # 1 -> 2: Unique ID format changed:
+    if version == 1:
+        config_entry.version = 2
+        hass.config_entries.async_update_entry(
+            config_entry, data=config_entry.data)
+        _LOGGER.info('Migration to version %s successful', version)
+
+    msg = """{0}
+            Please remove the Ambient PWS Integration and re-configure
+            [here](/config/integrations).""".format(reasons_to_re_add[version])
+
+    hass.components.persistent_notification.async_create(
+        title="Ambient PWS Requires Update",
+        message=msg,
+        notification_id='config_entry_migration'
+    )
+    return False
```

Comment:
This will log an error saying that migration failed and not setup the entry at all.  Instead of doing a notification, would it make more sense to remove the old entry and initiate a new config flow?  This will have a similar effect of showing a notification but will take the user right into the new flow and you could potentially pass some of the previous data values if they are still valid.

Reply:
This is essentially what SmartThings does because there is no migration path from v1 to v2: https://github.com/home-assistant/home-assistant/blob/dc722adbb53edc1f7b1b7bc5d5f0d7098cc7ab97/homeassistant/components/smartthings/__init__.py#L42-L60

Reply:
Fascinating! It's a great idea – very transparent to the user. I'll revise.

Reply:
With this in place, is the breaking change described in https://github.com/home-assistant/home-assistant/pull/25284 still necessary?

Reply:
If the `entity_id` won't change, then it's not a breaking change, but if they could, then yes. :)

Reply:
Got it – I won't have changed it, but a user customization could; I'll update the verbiage. Thanks!